mono/tests/appdomain-unload.cs

```diff
@@ -191,6 +225,32 @@ public class Tests
 	}
 	*/
 
+	public static void ThreadStart (object obj)
+	{
+		IRunnable runnable = (IRunnable)obj;
+
+		try {
+			runnable.Run ();
+		} catch (AppDomainUnloadedException) {
+			Console.WriteLine ("OK");
+		} catch (ThreadAbortException) {
+			throw new Exception ();
+		}
+	}
+
+	public static int test_0_unload_reset_abort () {
+		AppDomain domain = AppDomain.CreateDomain ("test_0_unload_reset_abort");
```

Comment:
Could you also add tests for the expected behaviour in case of AppDomainUnloadedException in the different cases?